interpreter/activation_test.go

```diff
@@ -53,3 +53,17 @@ func TestHierarchicalActivation(t *testing.T) {
 		t.Error("Activation failed to resolve child value of 'c'")
 	}
 }
+
+func TestActivation_NilValue(t *testing.T) {
+	a, _ := NewActivation(map[string]interface{}{"nil": nil})
```

Comment:
I think this is not solving panic at all. It was trigger only when passing nil pointer to some defined type. For example:
```
var str *string
a, _ := NewActivation(map[string]interface{}{"nil": str})
```

Reply:
Ah, well, two bugs, one PR then. :)

I've update the code to do the following things:

* Check for `nil` after the type-assertions for pointers within the type-adapting code.
* For `nil` pointers to simple values (including duration and timestamp), I've chosen
  to return the empty value to be consistent with the CEL spec.
* For `nil` pointers to aggregate values and types, the type adapters will return a
  `types.Err` value as the zero value is not easy to infer and can cause undefined
  behavior in later call sequences.

In general, `nil` pointers shouldn't be provided as input to the program. The PR to
address #210 will handle error recovery in the event that other `panic`s lurk in the
shadows.

Reply:
Hmm, actually special handling for the `nil` pointer types is proving problematic. I've opted to instead treat `nil` pointers as errors and return a `types.Err` value when they're encountered.